File: common/src/main/java/com/example/todo/infrastructure/user/UserMapper.java
package com.example.todo.infrastructure.user;

import com.example.todo.domain.user.User;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Select;

@Mapper
public interface UserMapper {

  @Select("select id, user_name, user_id, password, created_at from user where id = #{id}")
  User findById(Long id);

  @Select("select id, user_name, user_id, password, created_at from user where user_id = #{userId}")
  User findByUserId(String userId);
}

File: common/src/main/java/com/example/todo/infrastructure/user/UserMapper.kt
package com.example.todo.infrastructure.user

import com.example.todo.domain.user.User
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Select
import org.springframework.stereotype.Repository

@Mapper
@Repository
interface UserMapper {

  @Select("select id, user_name, user_id, password, created_at from user where id = #{id}")
  fun findById(id: Long): User?

  @Select("select id, user_name, user_id, password, created_at from user where user_id = #{userId}")
  fun findByUserId(userId: String): User?
}
